cmd/kubelet/app/options/options.go

```diff
@@ -317,9 +317,8 @@ func AddKubeletConfigFlags(fs *pflag.FlagSet, c *kubeletconfig.KubeletConfigurat
 	fs.Int32Var(&c.ImageGCLowThresholdPercent, "image-gc-low-threshold", c.ImageGCLowThresholdPercent, "The percent of disk usage before which image garbage collection is never run. Lowest disk usage to garbage collect to.")
 	fs.DurationVar(&c.VolumeStatsAggPeriod.Duration, "volume-stats-agg-period", c.VolumeStatsAggPeriod.Duration, "Specifies interval for kubelet to calculate and cache the volume disk usage for all pods and volumes.  To disable volume calculations, set to 0.")
 	fs.StringVar(&c.VolumePluginDir, "volume-plugin-dir", c.VolumePluginDir, "<Warning: Alpha feature> The full path of the directory in which to search for additional third party volume plugins")
-	fs.StringVar(&c.FeatureGates, "feature-gates", c.FeatureGates, "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
+	fs.Var(flag.MapStringBool(c.FeatureGates), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
```

Comment:
Does this actually work? If so then why do we do the stuff above (L312)?

Reply:
Does this work because of the defaulting?

Reply:
Yeah, it works, I can run the dynamic kubelet config tests with this patch, with the feature gate set via the command line. I pinged @smarterclayton about L312 a few days ago, he said it had been too long since he wrote it to really remember, but that it might have either been an early go convention before we switched to plfag/cobra or that we may have been initializing it lazily at one point, then refactored it out (but probably the former).

Reply:
I'm fairly certain if you removed the defaulting, you would get a nil pointer dereference and that's why that is there. But sounds good.

Reply:
Interesting point. I think that might be true. But shouldn't be an issue, since we'll always default before applying the flags.